doc/PROTOCOL-SEMANTICS.md

```diff
@@ -0,0 +1,268 @@
+aware of the common restriction that RPCs are unable to survive longer than the
+connection on which they exist.
+
+The exact details of how the communication is performed and higher level
+abstractions can change when necessary and are thus not a focus for this
+document. Although implementations SHOULD support the [HTTP/2
+transport](PROTOCOL-HTTP2.md) to provide a basis for interoperability.
+
+## Methods
+
+An RPC is performed on a Method. The Method has a name, defines the intended
+operation, the message types involved with the RPC, and the cardinality of
+those messages.  It does not define the endpoint to send the RPC to; this is
+normally handled by the Channel.
+
+A Method's intended operation is just normal documentation describing what a
+Method does, intended for a developer. gRPC itself is not generally aware of the
+intended operation.
+
+A Method has a request message type and a separate response message type. gRPC
+is only aware of these types well enough to serialize and deserialize them.
+Messages are considered opaque byte sequences of a known length to gRPC itself.
+
+A Method's request and response each have a cardinality: either one ("unary"),
+or zero to many (a "stream"). This produces four possible configurations which,
+for convenience, each have a name:
+
+|                       | **unary response** | **streaming response** |
+| --------------------- | ------------------ | ---------------------- |
+| **unary request**     | unary              | server-streaming       |
+| **streaming request** | client-streaming   | bidirectional (bidi)   |
+
+While we use the term "method," pedantically it is closer to "function" as it is
+not object oriented and there is no "receiver" involved (the `this` variable in
+many languages) other than the destination machine. gRPC is based on message
+passing, not object orientation.
+
+Related Methods are typically grouped into a Service. To gRPC, a Service is a
+group of methods that tend to be implemented together and that all share the
+Service's namespace. A Service is a higher-level abstraction and may not be
+present explicitly in all implementations. However, the namespace provided by a
+Service is a core distinguishing feature of its Methods; if two Methods have the
+same name but exist in different Services they must be considered distinct and
+not be confused. A Method name including its Service namespace prefix with a "/"
+separator is a "full method name".
+
+## Calls
+
+RPCs, or "Calls," are initiated by a client to a server, typically via a
+Channel. There may be multiple servers that _could_ have received the Call (as
+is common for load balancing), but only a single server may process an
+individual Call. Calls are assumed to be non-idempotent and may not be
+"replayed" except for when gRPC is explicitly informed it is safe to do so.
+
+Calls are natively two independent streams (i.e., full duplex bidirectional) of
```

Comment:
I recently pointed someone at this doc, and they misread this paragraph to mean that a gRPC bidi stream has a separate H2 stream for each direction.  When we eventually get around to finalizing this doc, we should try to clean up the language here to avoid this confusion.